sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala

```diff
@@ -125,19 +125,15 @@ abstract class CentralMomentAgg(child: Expression) extends ImperativeAggregate w
       mean += deltaN
       buffer.setDouble(meanOffset, mean)
 
-      if (momentOrder >= 2) {
-        m2 = buffer.getDouble(secondMomentOffset)
-        m2 += delta * (delta - deltaN)
-        buffer.setDouble(secondMomentOffset, m2)
-      }
+      m2 = buffer.getDouble(secondMomentOffset)
+      m2 += delta * (delta - deltaN)
+      buffer.setDouble(secondMomentOffset, m2)
 
-      if (momentOrder >= 3) {
```

Comment:
Those `if` branches are important to save computation for low-order statistics. Even we won't use `CentralMomentAgg` for second-order statistics, it is still good to keep them.